tests/test_phase5_rebuild_ete.py

```
@@ -0,0 +1,785 @@
+
+
+@pytest.fixture
+def lua():
+    rt = lupa.LuaRuntime(unpack_returned_tuples=False)
+    rt.execute(STUB_UI_LUA)
+    modules_path = str(MODULES).replace("\\", "/")
+    rt.execute(f'package.path = package.path .. ";{modules_path}/?.lua"')
+    return rt
+
+
+def _build_trace(lua, n=2000):
```

Ruff (0.15.9)

[warning] Missing return type annotation for private function `_build_trace`

(ANN202)